Add information about adding same variant to multiple lines in checkout (#479)
saleor/saleor-docs: docs/developer/checkout.mdx

The custom price can be set with the use of field `price` in the
[CheckoutLineInput](developer/api-reference/inputs/checkout-line-input.mdx)
in the following mutations:

- [checkoutCreate](developer/api-reference/mutations/checkout-create.mdx),
- [checkoutLinesAdd](developer/api-reference/mutations/checkout-lines-add.mdx) - when adding

>   a variant that is already added to checkout, the already existing line is overridden - the quantity is incremented and the price is updated.,

Reviewer: ```suggestion
  a variant that already exists in the checkout, the corresponding line gets overridden - the quantity is incremented and the price is updated.
```

Those two items will also be subtracted from the available quantity displayed to other users on store, giving them immediate feedback that item they are interested in is currently not available, preventing them from adding this item to their own checkouts.

> If user A completes their checkout within given time, temporary allocation will be converted into permanent one. But if they don't complete their stock within the time limit, their stock reservation will expire and the two items in their checkout will become available for other users to add to their own checkouts.

Reviewer: ```suggestion
If user A completes their checkout within the given time, temporary allocation will be converted into a permanent one. But if they don't complete their stock within the time limit, their stock reservation will expire and the two items in their checkout will become available for other users to add to their own checkouts.
```
